inference-engine/src/transformations/include/transformations/snippets/generator.hpp

```diff
@@ -0,0 +1,66 @@
+// Copyright (C) 2020 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <transformations_visibility.hpp>
+#include "ngraph_ops/snippets_isa.hpp"
+
+namespace ngraph {
+namespace snippet {
+
+using code = const uint8_t *;
+using RegInfo = std::pair<std::vector<size_t>, std::vector<size_t>>;
+
+TRANSFORMATIONS_API auto getRegisters(std::shared_ptr<ngraph::Node>& n) -> ngraph::snippet::RegInfo;
+
+class TargetMachine {
+};
+
+class Emitter {
+public:
+    Emitter(const std::shared_ptr<ngraph::Node>& n) {
+    }
+
+    virtual void emit(const std::vector<size_t>& in,
+                      const std::vector<size_t>& out,
+                      const std::vector<size_t>& pool = {},
+                      const std::vector<size_t>& gpr  = {}) const = 0;
+
+    // FIXME: remove in future
+    virtual void emit_table() {
+    }
+};
+
+// Generator interface
+class TRANSFORMATIONS_API Generator {
```

Comment:
Why do we need to have a generator in our transformation library?
Looks like it is a part of code which can be used only by CPU plugin.

Reply:
It's supposed to be target independent. Plugin just provides table for target dependent code generation.

I think either it should be separate directory for codegen related things or it should be part of transformation library. What do you think?

Reply:
From my point of view it should be a separate target, because majority of plugins uses the transformation library but I am not sure that all plugins will support codegen feature.

Reply:
what folder structure do you propose then? `inference-engine/src/snippets/`?

Reply:
Current master `inference_engine_transformations`:
```
2 979 776 # default build
1 849 272 # LTO
```

Current branch (merge with master):
```
3 307 456 # default build
2 103 224 # LTO
```

gcc 5.5.0

So, the binary increase is about 250 kb.. If the `snippets` module is not going to grow and grow, I think we can keep it as a part of transformations library..

Reply:
@ilya-lavrenov, I don't think it would explode, moreover I plan to remove debug out once I make build green 😄